scripts/anvil_tools/set_up_anvil_workspaces.py

```diff
@@ -0,0 +1,255 @@
+
+    if status_code == 404:  # if workspace does not exist
+        return False
+    elif status_code == 200:  # if workspace already exists
+        print(f"Workspace already exists with name: {project}/{workspace_name}.")
+        print(f"Existing workspace details: {json.dumps(response.json(), indent=2)}")
+        # make user decide if they want to update/overwrite existing workspace
+        update_existing_ws = input("Would you like to continue modifying the existing workspace? (Y/N)" + "\n")
+        return update_existing_ws.upper()
+    else:  # if other error
+        return response.json()
+
+
+def create_workspace(json_request, workspace_name, project="anvil-datastorage"):
+    """Create the Terra workspace with given authorization domain."""
+
+    # check if workspace already exists
+    status_ws_exists = check_workspace_exists(workspace_name, project)
+
+    # if workspace does not exist, create workspace
+    if not status_ws_exists:
+        # request URL for createWorkspace
+        uri = f"https://api.firecloud.org/api/workspaces"
+
+        # Get access token and and add to headers for requests.
+        # -H  "accept: application/json" -H  "Authorization: Bearer [token] -H  "Content-Type: application/json"
+        headers = {"Authorization": "Bearer " + get_access_token(), "accept": "application/json", "Content-Type": "application/json"}
+
+        # capture response from API and parse out status code
+        response = requests.post(uri, headers=headers, data=json.dumps(json_request))
+        status_code = response.status_code
+
+        # print success or fail message based on status code
+        if status_code != 201:
+            print(f"WARNING: Failed to create workspace with name: {workspace_name}. Check output file for error details.")
+            return response.json()
+        else:
+            print(f"Successfully created workspace with name: {workspace_name}.")
+            return status_code
+    elif status_ws_exists == "N":
+        return(f"{project}/{workspace_name} already existed. User selected not to overwrite. Try again with unique workspace name.")
+    # if workspace already exists, dont make workspace
+    else:
+        return status_ws_exists  # ("Y" or json with other errors)
+
+
+def make_create_workspace_request(workspace_name, auth_domain_name, project="anvil-datastorage"):
+    """Make the json request to pass into create_workspace()."""
+
+    # initialize empty dictionary
+    create_ws_request = {}
+
+    create_ws_request["namespace"] = project
+    create_ws_request["name"] = workspace_name
+    create_ws_request["authorizationDomain"] = [{"membersGroupName": f'{auth_domain_name}'}]
+    create_ws_request["attributes"] = {}
+    # TODO: set noWorkspaceOwner = True for data delivery workspaces - picard svc is the only owner
+    create_ws_request["noWorkspaceOwner"] = False
+
+    return create_ws_request
+
+
+def add_members_to_auth_domain(auth_domain_name):
```

Comment:
if this is only adding admins, can you rename to something like `add_anvil_admins_to_auth_domain` ? alternatively (this may be better but it's up to you) if you want to keep it general, then i'd include an input list of emails to add, and hardcode the anvil admin email one level up